ext/vscode/src/commands/down.ts

```diff
@@ -22,6 +22,7 @@ export async function down(context: IActionContext, selectedItem?: vscode.Uri |
     const purgeDelete: MessageItem = { title: vscode.l10n.t("Delete and Purge") };
 
     // If cancel is chosen or the modal is closed, a `UserCancelledError` will automatically be thrown, so we don't need to check for it
+    // External prompting applies but is not used here because the prompt is special--not an ordinary yes/no confirmation prompt
```

Comment:
@bwateratmsft  - We do have the `confirm` prompt today.  Do you think we should have `confirmWarning` or some `dangerous`/`warning` flag on the confirm message so that consumers can display things this way?

I suspect that like `init`, `down` is a place that over time we may add more confirmations - so it would be really great if we could get this on external prompting.

Reply:
I had sketched out a potential implementation where `confirm` could send in options, same as select and multi-select, with each option resulting in a button. We could do something like that to cover the purge prompt and potential future `down` prompts.